src/Func.cpp

```diff
@@ -1546,11 +1546,22 @@ Stage &Stage::reorder(const std::vector<VarOrRVar>& vars) {
     vector<Expr> &args = definition.args();
     vector<Expr> &values = definition.values();
 
-    // Check whether the operator is associative and commutative.
-    const auto &prover_result = prove_associativity(func_name, args, values);
-    const bool is_associative_and_commutative
-            = prover_result.associative() && prover_result.commutative();
+    bool is_associative_and_commutative = false;
 
+    bool any_rvars = false;
```

Comment:
I think we can just check whether there are any RVar in the input to this `reorder` function. There is no point to run the prover if a user is not trying to reorder two RVars.

Reply:
+1, going even further, it's only necessary if RVars are being reordered w.r.t. each other. A hack would be to only do this check if >= 2 RVars are being reordered, a more invasive change would be to only do this check when an error is about to be emitted.